File: 3236-smallest-missing-integer-greater-than-sequential-prefix-sum/3236-smallest-missing-integer-greater-than-sequential-prefix-sum.py
class Solution:
    def missingInteger(self, arr: List[int]) -> int:
        sum = arr[0]
        n = len(arr)

        for i in range(1,n):
            if arr[i] == arr[i-1] + 1:
                sum += arr[i]
            else: 
                break
        
        missing = sum
        found = True

        while found:
            found = False
            for i in range(0, n):
                if arr[i] == missing:
                    missing += 1
                    found = True
                    break
        
        return missing

File: 3236-smallest-missing-integer-greater-than-sequential-prefix-sum/3236-smallest-missing-integer-greater-than-sequential-prefix-sum.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingInteger = function(arr) {
    let sum = arr[0];
    let n = arr.length;

    for (let i = 1; i < n; i++) {
        if (arr[i] === arr[i - 1] + 1) {
            sum += arr[i];
        } else {
            break;
        }
    }

    let missing = sum;
    let found = true;

    while (found) {
        found = false;
        for (let i = 0; i < n; i++) {
            if (arr[i] === missing) {
                missing++;
                found = true;
                break;
            }
        }
    }
    return missing;
};